### PR TITLE
Fix ESLint

### PR DESCRIPTION
**Pull request by Nadinefiona**
#### What does this PR do?

This PR configures ESLint and a Git pre-commit hook to run tests and format the code base according to ESLint and Prettier rules. This ensures that all code committed to the repository adheres to our coding standards and passes all tests.

#### Description of Task to be completed?

- Install ESLint and Prettier.
- Configure ESLint and Prettier according to project standards.
- Install Husky for Git hooks.
- Configure a pre-commit hook to run ESLint and tests.

#### How should this be manually tested?

1. Make a change to the code that should trigger an ESLint error or a failing test.
2. Attempt to commit the change. The commit should be blocked, and the error messages should be displayed.
3. Fix the ESLint error or test failure.
4. Attempt to commit the change again. The commit should now succeed.

#### Any background context you want to provide?

This setup is crucial for maintaining code quality and preventing breaking changes from being committed. It ensures that all code adheres to our coding standards and passes all tests before it is merged into the main branch.

#### What are the relevant pivotal tracker/Trello stories?

- Story ID: 187300132

### CODE REVIEW

**Review by Mukunzijames — APPROVED** [review]

**Review by fab-ryan — COMMENTED** [review]
Rebase first and correct where we have ESLint errors

**Review by fab-ryan — CHANGES_REQUESTED** [review]
Rebase First

File: src/controllers/userController.ts [hunk 1]
```diff
@@ -4,14 +4,16 @@ import jwt from 'jsonwebtoken';
...
 import getDatabaseConfig from '../config/config.js';
 
 dotenv.config();
 
 interface UserCreationAttributes extends Omit<UserAttributes, 'id'> {}
 
 export const createUser = async (
+  // eslint-disable-next-line @typescript-eslint/ban-types
   req: Request<{}, {}, UserCreationAttributes>,
   res: Response
```
Comment (fab-ryan) [inline src/controllers/userController.ts]: Please here use type of object

File: src/middleware/validate.ts [hunk 2]
```diff
@@ -1,52 +1,56 @@
 import { NextFunction, Request, Response } from 'express';
-import { logger, sendResponse } from '../utils';
 import Joi from 'joi';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (fab-ryan) [inline src/middleware/validate.ts:3]: if we have unused vars remove it please

File: src/routers/index.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
 import { Router, Request, Response } from 'express';
```
Comment (fab-ryan) [inline src/routers/index.ts]: remove those unused vars

**Review by fab-ryan — COMMENTED** [review]
check well every possible out comes

File: src/routers/index.ts (hunk 3 above)
Comment (Nadinefiona) [reply src/routers/index.ts]: you meant i need to remove those  eslint comments?

**Review by Ruthuwamahoro — COMMENTED** [review]
Great Job

**Review by Ruthuwamahoro — APPROVED** [review]
Great job

**Review by Ruthuwamahoro — COMMENTED** [review]
First resolve that errors

File: src/controllers/loginController.ts [hunk 4]
```diff
@@ -41,25 +42,23 @@ const validatingUser = async (req: Request, res: Response) => {
   passport.authenticate(
     'local',
     { session: false },
-    async(err: any, user: any) => {
-        
-      //test if uer not found
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    async (err: any, user: any) => {
+      // test if uer not found
```
Comment (fab-ryan) [inline src/controllers/loginController.ts:47]: please avoid to use any type use User type from Model

**Review by fab-ryan — APPROVED** [review]

**Pull request merged**